Debug/Release build discrepancy in some conditions

**Describe the bug**

I have found a condition (number 2 in the code example) where compilation is ok in debug mode but fails in release mode.

```rust
    // ##################################################################
    // 1. Debug OK, Release OK
    <Show 
        when=move || true // dummy condition to reduce the example
        fallback=move || view! {
            <A href="/link1" on:click=on_nav_link_click>
                Link1
            </A>
        }
    >
        <p>some text</p>
    </Show>

    // ##################################################################
    // 2. Debug OK, Release Fails
    <Show 
        when=move || true // dummy condition to reduce the example
        fallback=move || view! {
            <A href="/link1" on:click=on_nav_link_click>
                Link1
            </A>
        }
    >
        <A href="/link2" on:click=on_nav_link_click> // <--- ADDED THIS
            Link2
        </A>
    </Show>

    // ##################################################################
    // 3. Debug OK, Release OK
    <Show 
        when=move || true // dummy condition to reduce the example
        fallback=move || view! {
            <A href="/link1" on:click=move |e| on_nav_link_click(e)> // <--- CHANGED THIS
                Link1
            </A>
        }
    >
        <A href="/link2" on:click=on_nav_link_click>
            Link2
        </A>
    </Show>
```

Compilation error (not extremely helpful to pinpoint the faulty line of code):

```
error[E0310]: the associated type `<impl leptos::IntoView as leptos::prelude::AddAnyAttr>::Output<(On<click, {closure@src/app.rs:240:29: 240:37}>,)>` may not live long enough
   --> src/app.rs:278:5
    |
278 | /     view! {
279 | |         // injects a stylesheet into the document <head>
280 | |         // id=leptos means cargo-leptos will hot-reload this stylesheet
281 | |         <Stylesheet id="leptos" href="/pkg/dj-trex.css"/>
...   |
479 | |         </Router>
480 | |     }
    | |     ^
    | |     |
    | |_____the associated type `<impl leptos::IntoView as leptos::prelude::AddAnyAttr>::Output<(On<click, {closure@src/app.rs:240:29: 240:37}>,)>` must be valid for the static lifetime...
    |       ...so that the type `<impl leptos::IntoView as leptos::prelude::AddAnyAttr>::Output<(On<click, {closure@src/app.rs:240:29: 240:37}>,)>` will meet its required lifetime bounds
    |
    = help: consider adding an explicit lifetime bound `<impl leptos::IntoView as leptos::prelude::AddAnyAttr>::Output<(On<click, {closure@src/app.rs:240:29: 240:37}>,)>: 'static`...
    = note: this error originates in the macro `view` (in Nightly builds, run with -Z macro-backtrace for more info)

```

**Leptos Dependencies**

```toml
leptos = { version = "0.8.0", features = ["nightly", "tracing", "rkyv"] }
leptos_router = { version = "0.8.0", features = ["nightly", "tracing"] }
leptos_meta = { version = "0.8.0" }
leptos_axum = { version = "0.8.0", optional = true }
```

**Next Steps**

- [ ] I will make a PR
- [ ] I would like to make a PR, but need help getting started
- [ ] I want someone else to take the time to fix this
- [x] This is a low priority for me and is just shared for your information

Oh year very sorry, I should have tried to reduce the example more. Turns you don't even need the click handler but this seems to happen only if I put the code block inside `<Router>` and not inside a route.

Full reproducible example:

1. start from axum SSR template (probably not necessary but that's what I did), with nightly or stable (both exhibit the behavior)
2. modify `app.rs` to be:

```rust
use leptos::prelude::*;
use leptos_meta::{provide_meta_context, MetaTags, Stylesheet, Title};
use leptos_router::{
    components::{Route, Router, Routes, A},
    StaticSegment,
};

pub fn shell(options: LeptosOptions) -> impl IntoView {
    view! {
        <!DOCTYPE html>
        <html lang="en">
            <head>
                <meta charset="utf-8"/>
                <meta name="viewport" content="width=device-width, initial-scale=1"/>
                <AutoReload options=options.clone() />
                <HydrationScripts options/>
                <MetaTags/>
            </head>
            <body>
                <App/>
            </body>
        </html>
    }
}

#[component]
pub fn App() -> impl IntoView {
    // Provides context that manages stylesheets, titles, meta tags, etc.
    provide_meta_context();

    view! {
        // injects a stylesheet into the document <head>
        // id=leptos means cargo-leptos will hot-reload this stylesheet
        <Stylesheet id="leptos" href="/pkg/leptos-discrep-mwe.css"/>

        // sets the document title
        <Title text="Welcome to Leptos"/>

        // content for this welcome page
        <Router>
            // ##################################################
            // ### This here: Debug builds OK, Release Fails
            <Show
                when=move || true
                fallback=move || view! {
                    <A href="/link1">Link1</A>
                }
            >
                <A href="/link2">Link2</A>
            </Show>
            // ##################################################

            <main>
                <Routes fallback=|| "Page not found.".into_view()>
                    <Route path=StaticSegment("") view=HomePage/>
                </Routes>
            </main>
        </Router>
    }
}

/// Renders the home page of your application.
#[component]
fn HomePage() -> impl IntoView {
    // Creates a reactive value to update the button
    let count = RwSignal::new(0);
    let on_click = move |_| *count.write() += 1;

    view! {
        <h1>"Welcome to Leptos!"</h1>
        <button on:click=on_click>"Click Me: " {count}</button>
        // ##################################################
        // If you put the <Show> block here though,
        // no discrepancy, both builds work
        // ##################################################
    }
}
```

3. `cargo leptos watch` -> OK, `cargo leptos watch --release` -> fails with:

```
error[E0310]: the opaque type `leptos_router::components::A<&str>::{opaque#0}` may not live long enough
  --> src/app.rs:31:5
   |
31 | /     view! {
32 | |         // injects a stylesheet into the document <head>
33 | |         // id=leptos means cargo-leptos will hot-reloa...
34 | |         <Stylesheet id="leptos" href="/pkg/leptos-disc...
...  |
58 | |         </Router>
59 | |     }
   | |     ^
   | |     |
   | |_____the opaque type `leptos_router::components::A<&str>::{opaque#0}` must be valid for the static lifetime...
   |       ...so that the type `impl leptos::IntoView` will meet its required lifetime bounds
   |
   = help: consider adding an explicit lifetime bound `leptos_router::components::A<&str>::{opaque#0}: 'static`...
   = note: this error originates in the macro `view` (in Nightly builds, run with -Z macro-backtrace for more info)
```

--- a/router/src/link.rs
+++ b/router/src/link.rs
@@ -116,7 +116,7 @@ pub fn A<H>(
     scroll: bool,
     /// The nodes or elements to be shown inside the link.
     children: Children,
-) -> impl IntoView
+) -> impl IntoView + 'static
 where
     H: ToHref + Send + Sync + 'static,
 {

